### ISSUE TITLE
Managed Identity Role assignment in Sentinel

### ISSUE THREAD

**Issue by c03vikas** [Azure/Azure-Sentinel#4227]
Most of my SOAR solution in Sentinel is using logic apps and many of them come with System assigned managed identities. I am facing this issue with my several logic apps where I try to assign any role via Azure portal, I get push notification of role assignment success but doesn't populate on page after multiple browser refresh. And it also complaints when I try to assign the role again. 
Moreover, the managed identity doesn't work in logic app and gives error even on simple actions such as sentinel incident update, even after giving a sentinel contributor role to MI. 

**To Reproduce**
1. Go to Identity section of any logic app. 
2. Click on 'Azure role assignments' permissions and click on Add role assignments 
3. Assign any role, I have tested and faced it with many roles and sentinel contributor being one if them.
4. See error - that role does get's assigned as can be seen in push notification but doesn't pop-up even after multiple refresh on page.
5. Also the logic app doesn't gives error every time it hits the action - update incident.
6. This problem can be seen with CLI as well. 

**Expected behavior**
A role should get assigned to MI and the logic app should run without any errors.

**Screenshots**
![image](https://user-images.githubusercontent.com/8371958/154893258-b3ab0bfd-a21f-478c-a411-0fb61b3973f6.png)

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser Chrome
 - Version 98.0.4758.102

**Additional context**
Some cases in my logic apps when I put a condition block true and false, and configure a same action to update incident on both sides and different severity to change. If condition's outcome is true then the logic app runs without any error but if it is false then it gives error "Incident ARM ID cannot be fetched" and app fails.

**Comment by BenjiSec** [comment]
Hi @c03vikas - When you apply role to managed identity it takes couple of minutes until it's shown in playbook interface. But the role is automatically applied and you can check it from subscription/resource group IAM page. You get error when you try again because role is already applied.

For the error that you are getting when using update incident, do you authorize connection with managed identity? 
Can you please share screenshot of complete action as you fill it with dynamic content and the error?

If issue is with Managed Identity, error would be that there is no enough privileges to run action. The error is important to understand what is the issue.

**Comment by c03vikas** [comment]
I agree with you that it might take some time after we apply the role.
Yes, I authorize the connection with managed identity.
I have deleted the logic app so, unable to attach the screenshot. 

I believe the issue is sometimes when I assign a role to system managed identity it doesn't get attached to it, even in the backend. Many of time this has happened that I assign a role and after few minutes and couple of refresh start showing the role and it works well, but in some rare cases it doesn't attach the role permission to that MI and even after waiting period of 1-2 days it doesn't show in portal. I had to reassign the role and currently I am using user assigned MI as a fix which works perfectly fine.    

Apologies that I could not get you a error message but in a single tree one branch of logic app was working and other wasn't that was my case, too bad I had to delete that resource. Thank you for your assistance.
